### PR TITLE
V2 password reset design

### LINKED ISSUE #4293: reset password needs v2 design
CleanShot 2022-09-08 at 16 17 34@2x

### PR DESCRIPTION
**Pull request by PeerRich**
haven't tested locally because I dont have a mailer installed.

can someone test locally? 

before:
CleanShot 2022-09-09 at 09 56 28@2x

Updated UI : 

https://user-images.githubusercontent.com/55134778/189358908-7157f6b6-4239-455e-a7c2-39ef0ffdf9ff.mp4

### CODE REVIEW

**Comment by PeerRich** [comment]
we should prob. also use our new TextInput component tho I am not quite sure how to use it, yet. @sean-brydon maybe can you help?

File: apps/web/pages/api/auth/forgot-password.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     });
 
     if (!maybeUser) {
-      return res.status(400).json({ message: "Couldn't find an account for this email" });
+      // Don't leak information about whether an email is registered or not
+      return res.status(200).json({ message: "Reset Requested" });
```
Comment (sean-brydon) [inline apps/web/pages/api/auth/forgot-password.ts:32]: On FE still redirect to success screen - even If the user isn't found

File: apps/web/pages/auth/forgot-password/[id].tsx [hunk 2]
```diff
@@ -103,102 +100,68 @@ export default function Page({ resetPasswordRequest, csrfToken }: Props) {
   }, [resetPasswordRequest]);
 
   return (
-    <div className="flex min-h-screen flex-col justify-center bg-gray-50 py-12 sm:px-6 lg:px-8">
```
Comment (sean-brydon) [inline apps/web/pages/auth/forgot-password/[id].tsx:106]: Wrap in auth container

File: apps/web/pages/api/auth/forgot-password.ts (hunk 1 above)
Comment (PeerRich) [reply apps/web/pages/api/auth/forgot-password.ts:32]: good

Comment (hariombalhara) [reply apps/web/pages/api/auth/forgot-password.ts:32]: Something like`If this email exists in our system, you should receive a Reset email.` is more accurate without leaking information.

**Review by hariombalhara — APPROVED** [review]
Looks good except one Text change.

**Comment by hariombalhara** [comment]
Fixing failing tests.

**Pull request merged**